src/components/ReportActionItem/MoneyRequestView.js

```diff
@@ -86,8 +86,10 @@ function MoneyRequestView({report, parentReport, shouldShowHorizontalRule, trans
 
     const hasReceipt = TransactionUtils.hasReceipt(transaction);
     let receiptURIs;
+    let hasErrors = false;
     if (hasReceipt) {
```

Comment:
I don't think we smartscan in those cases

Reply:
That seems reasonable, yes.

Reply:
I don't think I would return anything new from the server (especially since `isScanRequest ` isn't something stored in the database). The proper way to know if a receipt was smartscanned is if the receipt has one of the `CONST.IOU.RECEIPT_STATE` states. I think `TransactionUtils.hasReceipt(transaction);` should be updated to look at that instead. Can you look into doing that?

Reply:
Also I am thinking that we maybe need two functions...

- `TransactionUtils.hasAttachment()`
- `TransactionUtils.hasReceipt()` 

This would cover both cases and be more explicit for cases like this which are only meant to handle one case.

Reply:
Hm, the backend already has several methods that are similar to that. I still don't know that I am following this discussion completely but from the way I see it, the FE code for `isScanRequest()` needs to be refactored to account for the difference between a smartscan request and a manual request with attachment.

What does the backend have to do with it?

Reply:
The check for errors is done only when there is receipt and that caused #34997